pkg/controller/controller.go

```diff
@@ -40,8 +40,8 @@ type CertificateApprover struct {
 	MachineRestCfg   *rest.Config
 	MachineNamespace string
 
-	Config   ClusterMachineApproverConfig
-	APIGroup string
+	Config    ClusterMachineApproverConfig
+	APIGroups []string
```

Comment:
This should be group versions, what about using a [schema.GroupVersion](https://pkg.go.dev/k8s.io/apimachinery/pkg/runtime/schema#GroupVersion) as well, this would make it easier to grab the group versions out, and, we can use the parse function in the main to validate the flag options
```suggestion
	APIGroupVersions []schem.GroupVersion
```